### PR TITLE
chore(config): sets up "release" PR template and expands standard one

### PR DESCRIPTION
**Pull request by bkrem**
# Description

- Expands standard PR template
- Adds dedicated `release` PR template to ensure due diligence with downstream stakeholders (e.g. RN, web3modal)

### CODE REVIEW

File: .github/pr_template_release.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] **Canary Version Testing**
+
+  - [ ] I have thoroughly tested the release using a canary version: `v2...`
+  - [ ] The canary version has been verified to work as expected.
+  - [ ] All major features and changes have been tested and validated.
+
+- [ ] **Web3Modal Team Testing**
+
+  - [ ] The Web3Modal team has tested the release for compatibility and functionality.
+  - [ ] The package integrates seamlessly with Web3Modal (if relevant).
```
Comment (xzilja) [inline .github/pr_template_release.md]: Perhaps this could be rephrased in a way like: "The package is backwards compatible with Web3Modal (if relevant)."

**Review by xzilja — APPROVED** [review]
👌 Really thorough, nice

File: .github/pr_template_release.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## Additional Notes
+
+Include any additional information, concerns, or considerations related to the release that might be important for the reviewers to be aware of.
+
+## Definition of Done
+
+- [ ] The release has been tested using a canary version.
+- [ ] The release has been reviewed and approved by the Web3Modal team (if relevant).
+- [ ] The release has been reviewed and approved by the React Native team (if relevant).
+- [ ] All necessary documentation, including API changes or new features, has been updated.
```
Comment (ignaciosantise) [inline .github/pr_template_release.md:44]: we could add a section with related links for PRs with docs/api/other updates

File: .github/pr_template_standard.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+Please include a brief summary of the change.
+
+## Type of change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
```
Comment (JonathanConn) [inline .github/pr_template_standard.md:12]: Should we add a section for the 'chore' type? I have seen a few of those in the commit history

**Review by Cali93 — APPROVED** [review]

File: .github/pr_template_standard.md (hunk 3 above)
Comment (bkrem) [reply .github/pr_template_standard.md:12]: Good call thank you 👍

File: .github/pr_template_release.md (hunk 2 above)
Comment (bkrem) [reply .github/pr_template_release.md:44]: Added a "Related Issues and Pull Requests" section 👍

File: .github/pr_template_release.md (hunk 1 above)
Comment (bkrem) [reply .github/pr_template_release.md]: Rephrased and also aligned the bullet points more between web3modal and RN/web3modal-RN QA sections 👌

**Pull request merged**